.github/workflows/rust-test.yml

```diff
@@ -1,23 +1,46 @@
-name:                   Rust backend test
+name:                         Rust test
 
 on:
   pull_request:
   push:
     branches:
       - master
       - stable
+
 jobs:
   check:
-    name:               Check
-    runs-on:            ubuntu-latest
+    name:                     Cargo nextest
+    runs-on:                  ubuntu-latest
     steps:
-      - name:           Checkout sources
-        uses:           actions/checkout@v2
+
+      - name:                 Cancel Previous Runs
+        uses:                 styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token:       ${{ github.token }}
+
+      - name:                 Checkout sources
+        uses:                 actions/checkout@v2.4.0
+        with:
+          fetch-depth:        50
+          submodules:         'recursive'
+
+      - name:                 Install Rust stable toolchain
+        uses:                 actions-rs/toolchain@v1.0.7
         with:
-          fetch-depth:  50
-      - name:           Checkout submodules
-        shell:          bash
-        run:            git submodule update --init --recursive
-      - name:           Rust build
-        shell:          bash
-        run:            cargo test
+          profile:            minimal
+          toolchain:          stable
+          override:           true
+
+      - name:                 Install cargo-nextest
+        uses:                 baptiste0928/cargo-install@v1
+        with:
+          crate:              cargo-nextest
+          version:            0.9
+
+      - name:                 Rust Cache
+        uses:                 Swatinem/rust-cache@v1.3.0
+        with:
+          working-directory:  .
+
+      - name:                 cargo nextest
+        run:                  cargo nextest run --retries 2
```

Comment:
Why retries `2`? This is a cryptolib, nothing should be flaky. And if the runner fails, this would not cover it.